library/x509write_csr.c

```diff
@@ -89,7 +89,8 @@ int mbedtls_x509write_csr_set_key_usage( mbedtls_x509write_csr *ctx, unsigned ch
 
     c = buf + 4;
 
-    if( ( ret = mbedtls_asn1_write_bitstring( &c, buf, &key_usage, 7 ) ) != 4 )
+    ret = mbedtls_asn1_write_named_bitstring( &c, buf, &key_usage, 7 );
+    if( ret != 4 )
```

Comment:
Fixing the bug in `mbedtls_asn1_write_bitstring()` leads to the following annoying problem: If a user sets the `MBEDTLS_X509_KU_ENCIPHER_ONLY` flag (bit 8) in the `key_usage` parameter (which according to the documentation should be ok), the previous code would add it in the CSR even though `asn1_write_bitstring()` was called with bitlength `7`, because of the missing masking of unused bits. Therefore, setting `MBEDTLS_X509_KU_ENCIPHER_ONLY` worked even though it was not supposed to. I don't know why we don't support `MBEDTLS_X509_KU_ENCIPHER_ONLY` and `MBEDTLS_X509_KU_DECIPHER_ONLY` (see https://github.com/ARMmbed/mbedtls/issues/2047), but to at least not break usage that worked before (by accident), I think we should write `8` bits here.